docs/concepts/components/callbacks.md

```diff
@@ -2,27 +2,92 @@
+let cb = link.callback(|text: String| Msg::Text(text));
+
+// The previous line is needlessly verbose to make it clearer.
+// It can be simplified it to this:
+let cb = link.callback(Msg::Text);
 
-### send\_message
+// Will send `Msg::Text("Hello World!")` to the component.
+cb.emit("Hello World!".to_owned());
+```
 
-Sends a message to the component immediately after the current loop finishes, causing another update loop to initiate.
+### batch_callback
 
-### batch\_callback
+Create a callback that will send a batch of messages to the component when it is executed.
+The difference to `callback` is that the closure passed to this method doesn't have to return a message.
+Instead, the closure can return either `Vec<Msg>` or `Option<Msg>` where `Msg` is the component's message type.
 
-Registers a callback that sends a batch of many messages at once when it is executed. If any of the messages cause the component to re-render, the component will re-render after all messages in the batch have been processed. A `Fn(IN) -> Vec<COMP::Message>` is provided and a `Callback<IN>` is returned.
+`Vec<Msg>` is treated as a batch of messages and uses `send_message_batch` under the hood.
+
+`Option<Msg>` calls `send_message` if it is `Some`. If the value is `None`, nothing happens.
+This can be used in cases where, depending on the situation, an update isn't required.
+
+This is achieved using the `SendAsMessage` trait which is only implemented for these types.
+You can implement `SendAsMessage` for your own types which allows you to use them in `batch_callback`.
+
+Like `callback`, this method also has a `FnOnce` counterpart, `batch_callback_once`.
+The same restrictions apply as for `callback_once`.
 
 ## Callbacks
 
 _\(This might need its own short page.\)_
 
-Callbacks are used to communicate with services, agents, and parent components within Yew. They are just a `Fn` wrapped by an `Rc` to allow them to be cloned.
+Callbacks are used to communicate with services, agents, and parent components within Yew. They are just an `Fn` wrapped by an `Rc` to allow them to be cloned.
 
 They have an `emit` function that takes their `<IN>` type as an argument and converts that to a message expected by its destination. If a callback from a parent is provided in props to a child component, the child can call `emit` on the callback in its `update` lifecycle hook to send a message back to its parent. Closures or Functions provided as props inside the `html!` macro are automatically converted to Callbacks.
 
+Simple usage of callbacks can look something like this:
+
+```rust
+let onclick = self.link.callback(|_| Msg::Clicked);
+html! {
+    <button onclick=onclick>{ "Click" }</button>
+}
+```
+
+The callback is always given some value. For example, the `onclick` handler will pass a `MouseEvent`. The handler can then decide what kind of message the event should resolve to. This message is scheduled for the next update loop unconditionally.
```

Comment:
```suggestion
The function passed to `callback` must always take a parameter. For example, the `onclick` handler requires a function which takes a parameter of type `MouseEvent`. The handler can then decide what kind of message the should be sent to the component. This message is scheduled for the next update loop unconditionally.
```

Reply:
typo: "what kind of message **the** should be"

Reply:
Thanks for the catch 😄!